packages/store/src/usersSlice/thunks.ts

```diff
@@ -746,30 +867,40 @@ export const getTokens = createAsyncThunk(`${sliceName}/getTokens`, (_, { dispat
 
 const ONE_YEAR = 31536000;
 
-export const generateToken = createAsyncThunk(`${sliceName}/generateToken`, ({ expiresIn = ONE_YEAR, name }, { dispatch }) =>
-  GeneralApi.post(`${useradmApiUrl}/settings/tokens`, { name, expires_in: expiresIn })
-    .then(({ data: token }) => Promise.all([dispatch(getTokens()), token]))
-    .catch(err => commonErrorHandler(err, 'There was an error creating the token:', dispatch))
+export const generateToken = createAppAsyncThunk(
+  `${sliceName}/generateToken`,
+  ({ expiresIn = ONE_YEAR, name }: { expiresIn?: number; name: string }, { dispatch }) =>
+    GeneralApi.post(`${useradmApiUrl}/settings/tokens`, { name, expires_in: expiresIn })
+      .then(({ data: token }) => Promise.all([dispatch(getTokens()), token]))
+      .catch(err => commonErrorHandler(err, 'There was an error creating the token:', dispatch))
 );
 
-export const revokeToken = createAsyncThunk(`${sliceName}/revokeToken`, (token, { dispatch }) =>
+export const revokeToken = createAppAsyncThunk(`${sliceName}/revokeToken`, (token: { id: string }, { dispatch }) =>
```

Comment:
There is `import type { PersonalAccessToken } from '@northern.tech/store/api/types/MenderTypes';` already.
```suggestion
export const revokeToken = createAppAsyncThunk(`${sliceName}/revokeToken`, (token: PersonalAccessToken, { dispatch }) =>
```

Reply:
That's unfortunate that we pass the whole object to the function that clearly requires only id (string). But now we have to live with it I guess

Reply:
The full type is more specific than a string + the thunk is called `revokeToken` (unlike say `revokeTokenById`), while having access to the token in question. Maybe you can elaborate why you think it's unfortunate?